src/java.base/share/classes/sun/util/PropertyResourceBundleCharset.java

```diff
@@ -73,7 +68,7 @@ public CharsetEncoder newEncoder() {
 
     private final class PropertiesFileDecoder extends CharsetDecoder {
 
-        private CharsetDecoder cdUTF_8 = UTF_8.INSTANCE.newDecoder()
+        private final CharsetDecoder cdUTF_8 = UTF_8.INSTANCE.newDecoder()
```

Comment:
Can be static as well.

Reply:
I'm not sure. Why do you think so? CharsetDecoder is not thread-safe.

Reply:
UTF8 decoder does not perform any internal state mutation during decoding; in addition, if this field is static final, the decoder's error actions are published when the class is initialized, while publishing in a final field does not guarantee the internal state of the charset is visible to other threads.

Reply:
> UTF8 decoder does not perform any internal state mutation during decoding; 

Are you sure? I think `CharsetDecoder::decode` will modify the `status` field.

Reply:
right, it does have a `state` field.

Reply:
CharsetDecoder is specified to be not safe for use by multiple concurrent threads, would be too fragile to assume behavior of a specific implementation.